base_layer/core/src/base_node/sync/horizon_state_sync/synchronizer.rs

```diff
@@ -301,6 +302,10 @@ impl<'a, B: BlockchainBackend + 'static> HorizonStateSynchronization<'a, B> {
 
             kernel_hashes.push(kernel.hash());
 
+            // TODO: We mix u32 and u64 for the mmr length. This comes down to the use of a 32-bit croaring Bitmap.
+            //       Suggest should use u64 externally (u64 is in the header) and error on the database calls if they
+            //       are > u32::MAX. Remove the clippy exception once fixed.
```

Comment:
The fix for the u64 bitmap is to add an array of bitmaps, I don't think it's as bad as you say, since when we get close to that number, we will need to add another element into the array